### PR TITLE
Add squad marker decorator values to junit item properties

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #6075 

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -185,8 +185,10 @@ def pytest_collection_modifyitems(session, items):
     # Add squad markers to each test item based on filepath
     for item in items:
         # check, if test already have squad marker manually assigned
-        if any(map(lambda x: "_squad" in x.name, item.iter_markers())):
-            continue
+        for marker in item.iter_markers():
+            if "_squad" in marker.name:
+                squad = marker.name.split("_")[0]
+                item.user_properties.append(("squad", squad))
```
Comment (ebenahar) [inline tests/conftest.py:191]: what if `_squad` not in the item's markers?

**Review by petr-balogh — DISMISSED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (clacroix12) [reply tests/conftest.py:191]: If there is no `_squad` in the item's existing markers we move onto the code below that will add the marker and property based on the [constant map](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/constants.py#L1540-L1558) we have defined for squads.

**Comment by clacroix12** [comment]
Took a look at the report from the PR test and I noticed that we capitalize the squad value when applying from the map. I want to match this behavior so we will need to call capitalize() when adding from the markers.

We also probably want to break out of the loop once we find the `_squad` marker as well as skip the for loop that adds markers/properties based on the path pattern.

**Review by vavuthu — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Comment by clacroix12** [comment]
Tier1 junit xml: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/jnk-pr6233b3212/jnk-pr6233b3212_20220728T173533/logs/test_results_1659029636.xml

All of the squad markers appear to be accurate. I checked a few specific test cases to make sure the correct marker appeared and went through each squad marker to make sure that there were no duplicates or multiple markers on a single test case. If there are no issues with the results here I think we can consider this verified. 

```
<testcase classname="tests.ui.test_pvc_ui.TestPvcUserInterface" name="test_create_resize_delete_pvc[ocs-storagecluster-cephfs-ReadWriteMany-2-Filesystem]" file="tests/ui/test_pvc_ui.py" line="29" time="398.792">
  <properties>
    <property name="squad" value="Green"/>
  </properties>
  ...
</testcase>
```

**Pull request merged**